docs/infrastructure/workers/index.md

```diff
@@ -47,4 +47,6 @@ Default pools attached to cloud targets allow co-location of workers and targets
 
 ## Multiple Projects Run Simultaneously on Workers
 
-The built-in worker can run many steps simultaneously.  For example, the task cap determines how many deployments can run simultaneously and those deployments could run many steps concurrently on the built-in worker.  External workers are the same.  External workers will run steps from different projects simultaneously.  This means external workers keep the behavior of the built-in worker, including that a step doesn't have exclusive access to the worker, which could allow one project to access the working folder of another project.
+The built-in worker can run many steps simultaneously.  For example, the [task cap](/docs/support/increase-the-octopus-server-task-cap.md) determines how many deployments can run simultaneously and those deployments could run many steps concurrently on the built-in worker.  External workers are the same.  External workers will run steps from different projects simultaneously.  This means external workers keep the behavior of the built-in worker, including that a step doesn't have exclusive access to the worker, which could allow one project to access the working folder of another project.
+
+The number of parallel worker invocations any one deployment task will make is controlled by the [Octopus System Variable](/docs/deployment-process/variables/system-variables.md) `Octopus.Action.MaxParallelism`.  For example, if your server task cap is 2 and the `Octopus.Action.MaxParallelism` variable is at its default value of 10, then you could be running 20 instances of the built-in worker concurrently (it's actually more because there could also be parallel steps within those deployment tasks).  If external workers are added to the default pool, then the workload is shared across those workers: a single worker will get the whole workload that the built-in worker used to do, two workers might get half each, etc.  Remember that in an HA setup each node has a task cap and can invoke a built-in worker locally, so you'll need to provision workers to at least match your server nodes, otherwise, you'll be asking each worker to do the sum of what the HA nodes were previously doing.
```

Comment:
Oh no! I didn't realised we'd further overloaded this variable with more meanings. I think we should change that design.

Reply:
Or at least talk it out! :)

Reply:
It's not really overloaded further.  It's just the same options as we always had.  In terms of how many worker invocations can run at once it's equivalent - just that where the workers are running has changed.

I put up an issue about splitting out the meanings of this var https://github.com/OctopusDeploy/Issues/issues/4667 and this one about controlling worker parallelism https://github.com/OctopusDeploy/Issues/issues/4815

It'll be quite the thing to change to allow something like task caps specified on workers, or task caps specified per pool (meaning the workers in this pool each have a task cap of X).  There'd potentially be complications around how we spin up the parallel using octo parallel (cause that's way up in DeploymentTaskController, but at the moment we don't make worker choices till way down in the ActionDispatcher) and we might have to consider how we distribute packages otherwise the workers with packages will end up being a bottleneck.  But the tricky bit might be the waiting cause we have to implement some sort of wait queue for threads waiting for workers, but then that'd have to be responsive to cancellation, potential reordering if we go with task priorities, etc.  

I think we should look at it, but I expect it will be subtle to get really robust.

Reply:
Of course! I see what you mean about the variable: we've just moved the location from the Octopus Server to a Worker, and the variable does just what it did before. Let's take this conversation offline. :)